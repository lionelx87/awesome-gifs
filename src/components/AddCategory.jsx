import PropTypes from 'prop-types'
import { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    // const onInputChange = (event) => {
    //     setInputValue(event.target.value);
    // };

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length < 1) return;
        // setCategories( categories => [ ...categories, inputValue ] );
        onNewCategory(inputValue.trim().toLocaleLowerCase());
        setInputValue('');
    }

    return (
        // <div>Add Category</div>
        <form onSubmit={ onSubmit } aria-label="form">
            <input 
                type="text"
                className="animate__animated animate__backInRight"
                placeholder="Look for that amazing gif!"
                value={ inputValue }
                // onChange={ (event) => onInputChange(event) }
                onChange={ onInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
};