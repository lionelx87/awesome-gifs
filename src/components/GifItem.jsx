import PropTypes from 'prop-types'
export const GifItem = ({ title, url, id }) => {

  const onClick = () => {
    navigator.clipboard.writeText(url);
  };
  
  return (
    <div className="card animate__animated animate__zoomInDown" onClick={ onClick }>
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}