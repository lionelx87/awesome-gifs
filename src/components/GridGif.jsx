import PropTypes from 'prop-types'
import { Fragment, useState, useEffect } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GridGif = ({ category }) => {

    const { images, loading } = useFetchGifs( category );

    return (
        <Fragment>
            <h3>{category}</h3>
            {
                // loading ? (<h2>Cargando...</h2>) : null
                loading && (<h2>Cargando...</h2>)
            }
            <div className="card-grid">
                {
                    images.map( (img) => (
                        <GifItem 
                            key={ img.id }
                            // image={ img }
                            // title={ img.title }
                            { ...img }
                        />
                    ))
                }
            </div>
        </Fragment>
    );
};

GridGif.propTypes = {
    category: PropTypes.string.isRequired
}