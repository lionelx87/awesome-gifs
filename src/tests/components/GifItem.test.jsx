import { render, screen } from '@testing-library/react'
import { GifItem } from '../../components/GifItem';


describe('Test in GifItem', () => {

    const title = "title";
    const url = "https://dragon-ball.jpg/";

    test('Should match the snapshot', () => {
        const { container } = render(<GifItem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();
    });

    test('Should display a valid url and alt attribute', () => {
        render(<GifItem title={ title } url={ url } />);
        // screen.debug();
        const { src, alt } = screen.getByRole('img');
        // expect( screen.getByRole('img').src).toBe( url );
        // expect( screen.getByRole('img').alt).toBe( title );
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Should display the title in the component', () => {
        render(<GifItem title={ title } url={ url } />);
        expect(screen.getByText(title)).toBeTruthy();
    });

});