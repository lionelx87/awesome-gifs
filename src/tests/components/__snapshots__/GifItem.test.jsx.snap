// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifItem Should match the snapshot 1`] = `
<div>
  <div
    class="card animate__animated animate__zoomInDown"
  >
    <img
      alt="title"
      src="https://dragon-ball.jpg/"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
